
// From https://github.com/ValvePython/steam/blob/26166e047b66a7be10bdf3c90e2e14de9283ab5a/protobufs/content_manifest.proto#L32

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

message ContentManifestPayload {
  message FileMapping {
    message ChunkData {
      optional bytes sha = 1;
      optional fixed32 crc = 2;
      optional uint64 offset = 3;
      optional uint32 cb_original = 4;
      optional uint32 cb_compressed = 5;
    }
    optional string filename = 1;
    optional uint64 size = 2;
    optional uint32 flags = 3;
    optional bytes sha_filename = 4;
    optional bytes sha_content = 5;
    repeated ChunkData chunks = 6;
    optional string linktarget = 7;
  }
  repeated FileMapping mappings = 1;
}

message ContentManifestMetadata {
  optional uint32 depot_id = 1;
  optional uint64 gid_manifest = 2;
  optional uint32 creation_time = 3;
  optional bool filenames_encrypted = 4;
  optional uint64 cb_disk_original = 5;
  optional uint64 cb_disk_compressed = 6;
  optional uint32 unique_chunks = 7;
  optional uint32 crc_encrypted = 8;
  optional uint32 crc_clear = 9;
}

message ContentManifestSignature {
  optional bytes signature = 1;
}